src/levanter/layers/attention.py

```diff
@@ -1474,15 +1477,31 @@ def init(config: AttentionConfig, *, key) -> "Attention":
             key=k_q,
             use_bias=use_bias,
             out_first=True,
+            reparam_cls=mup.OutputLinearMup if use_mup else mup.LinearStandardParam,  # Q out is not Width Dependent
         )
         k_proj = hnn.Linear.init(
-            In=config.Embed, Out=(config.KVHeads, config.HeadSize), key=k_k, use_bias=use_bias, out_first=True
+            In=config.Embed,
+            Out=(config.KVHeads, config.HeadSize),
+            key=k_k,
+            use_bias=use_bias,
+            out_first=True,
+            reparam_cls=mup.OutputLinearMup if use_mup else mup.LinearStandardParam,  # K out is not Width Dependent
```

Comment:
Wait, this is wrong right? either n_head or d_head will tend to infinity,

Reply:
<img width="967" height="173" alt="image" src="https://github.com/user-attachments/assets/14a455c4-2f7c-415f-b8c0-2b4d1c4a339f" />

Reply:
If n_head tends to infinity, then the same argument of output layers applies to each linear map. If d_head tends to infinity the argument is mildly different (happy to remove it), but the same muP factor should be used due to the correlation.

Reply:
I read that as the **attention** operation not q_proj and k_proj?

In the mup repo, [they initialize k_proj the same as o_proj](https://github.com/microsoft/mup/blob/main/examples/Transformer/model.py#L296-L304). Interestingly, they do zero-initialize q_proj, which is from Appendix D.2.

Reply:
(if @WhenWen and you both agree, I'll relent)

Reply:
ah nvm I'm dumb